core/primitives/src/errors.rs

```diff
@@ -1205,3 +1205,53 @@ impl From<near_vm_runner::logic::errors::FunctionCallError> for FunctionCallErro
         }
     }
 }
+
+#[cfg(feature = "new_epoch_sync")]
+pub mod epoch_sync {
+    use near_primitives_core::hash::CryptoHash;
+    use near_primitives_core::types::EpochHeight;
+    use std::fmt::{Debug, Display};
+
+    #[derive(Eq, PartialEq, Clone, strum::Display)]
+    pub enum EpochSyncHashType {
+        LastFinalBlock,
+        FirstEpochBlock,
+        NextEpochFirstBlock,
+        Other,
+    }
+
+    #[derive(Eq, PartialEq, Clone, thiserror::Error)]
+    pub enum EpochSyncInfoError {
+        HashNotFound { hash: CryptoHash, hash_type: EpochSyncHashType, epoch_height: EpochHeight },
+        ShortEpoch { epoch_height: EpochHeight },
+    }
+
+    impl EpochSyncInfoError {
```

Comment:
You may skip this in favor of `thiserror` macros, see the suggestions above.
The verbose error description is good, but a derived `Debug` implementation should suffice.